drafts/prometheus-alertmanager.adoc

```diff
@@ -0,0 +1,250 @@
+
+=== Groups
+If you have a network of systems that goes down, you probably don't want to receive an alert for every single instance - instead, it'd be preferable to get one alert that encapsulates all the other ones.
+
+In your `alertmanager.yml`, under `route`, you can group your alerts by label name:
+```
+route:
+  group_by: [ alerttype ]
+  group_wait: 10s
+  group_interval: 10s
+  repeat_interval: 5m
+  receiver: 'default_receiver'
+```
+
+The alerts will be grouped by `alerttype`, and the group will only send one alert with all of the information.
+
+image::/img/blog/prometheusAM_alertmanager_grouping.png[Alerts grouped in Alertmanager UI, align="left"]
+
+=== Inhibition
+For scenarios where you have multiple alerts that convey the same information, inhibiting your alerts can be useful. For example, if you have one alert that detects when 50% of your memory heap is used, and another alert for 90% of memory heap being used, there's no reason to send out alerts for the 50% problem.
+
+In your `alertmanager.yml`, add the following under `inhibition_rules`:
+```
+- source_match:
+    severity: 'critical'
+  # The alert that gets muted
+  target_match:
+    severity: 'warning'
+  # Both source and target need to have the same value to inhibit the alert
+  equal: [ 'alerttype' ]
+```
+The alert that has the label `severity: warning` (the target) will not be sent if there is an alert with the label `severity: critical` (the source). Both alerts must have the same value for the label `alerttype`. In our scenario, the alert `heapUsageAbove50%` will be inhibited if `heapUsageAbove90%` is firing at the same time.
+
+image::/img/blog/prometheusAM_alerts_firing.png[Alerts firing in Prometheus UI, align="left"]
+
+image::/img/blog/prometheusAM_slack_alert_inhibited.png[Slack alert for inhibited alert, align="left"]
+
+If we change the `alerttype` to be different values, the inhibition rule no longer matches, and both alerts will be sent. You can try it out by making the two `alerttype` labels different.
+
+== Silencing alerts
+Sometimes, you need to temporarily stop receiving alerts. For example, if you need to take your server down temporarily for maintenance, you don't want to receive any false positives. To do that, you can silence your alerts in the Alertmanager UI, under the `Silences` tab:
+
+image::/img/blog/prometheusAM_alertmanager_silences.png[Silencing Alerts in Alertmanager UI, align="left"]
+
+The matchers can be any metadata from your rules, e.g. labels, annotations, rule group name, etc.
+
+== Next steps
+
+Now that you have a basic configuration of the Prometheus Alertmanager set up, you can play with Prometheus rules to create a more comprehensive alerting system. You can also customize the messages to be more informative by using the Go templating system.
+
+Need some inspiration? Take a look at our link:https://github.com/jennifer-c/openliberty-alertmanager[a sample configuration for Open Liberty.]
```

Comment:
EDIT:
Take a look at our link:https://github.com/jennifer-c/openliberty-alertmanager[sample configuration for Open Liberty.]

(Removed "a")